pennylane/estimator/templates/subroutines.py

```diff
@@ -1998,6 +2005,172 @@ def resource_params(self) -> dict:
+        """
+
+        x = resource_rep(qre.X)
+        cnot = resource_rep(qre.CNOT)
+        l_elbow = resource_rep(qre.TemporaryAND)
+        r_elbow = resource_rep(qre.Adjoint, {"base_cmpr_op": l_elbow})
+
+        gate_cost = []
+
+        if num_data_blocks > 1:
+            gate_cost.append(
+                GateCount(x, count * (2 * (num_data_blocks - 2) + 1))
+            )  # conjugate 0 controlled toffolis + 1 extra X gate from un-controlled unary iterator decomp
+            gate_cost.append(
+                GateCount(
+                    cnot,
+                    count * (num_data_blocks - 2) + count * num_bit_flips,
+                )  # num CNOTs in unary iterator trick + each unitary in the select is just a CNOT
+            )
+            gate_cost.append(GateCount(l_elbow, count * (num_data_blocks - 2)))
+            gate_cost.append(GateCount(r_elbow, count * (num_data_blocks - 2)))
+
+        else:
+            gate_cost.append(
+                GateCount(
+                    x, count * num_bit_flips
+                )  # each unitary in the select is just an X gate to load the data
+            )
+        return gate_cost
+
+    @classmethod
+    def _ctrl_S(cls, num_ctrl_wires: int, count: int = 1) -> list[GateCount]:
+        """Constructs the control-S subroutine as defined in Figure 8 of
+        `arXiv:1092.02134 <https://arxiv.org/abs/1902.02134>_` excluding the initial ``X`` gate.
+
+        Args:
+            num_ctrl_wires (int): The number of control wires.
+            count (int): The number of times to repeat the subroutine.
+
+        Returns:
+            list[:class:`~.pennylane.estimator.resource_operator.GateCount`]: The resource decomposition of the control- :math:`S` subroutine.
+        """
+        num_ctrl_swaps = 2**num_ctrl_wires - 1
+        return [qre.GateCount(qre.resource_rep(qre.CSWAP), count * num_ctrl_swaps)]
+
+    @classmethod
+    def _ctrl_S_adj(cls, num_ctrl_wires: int, count: int = 1) -> list[GateCount]:
+        r"""Constructs the control-S^adj subroutine as defined in Figure 10
+        of `arXiv:1092.02134 <https://arxiv.org/abs/1902.02134>_` excluding the terminal ``X`` gate.
+
+        Args:
+            num_ctrl_wires (int): The number of control wires.
+            count (int): The number of times to repeat the subroutine.
+
+        Returns:
+            list[:class:`~.pennylane.estimator.resource_operator.GateCount`]: The resource decomposition of the control- :math:`S^{\dagger}` subroutine.
+
+        """
+        h = qre.resource_rep(qre.Hadamard)
+        cz = qre.resource_rep(qre.CZ)
+        cnot = qre.resource_rep(qre.CNOT)
+
+        num_ops = 2**num_ctrl_wires - 1
+        return [
+            qre.GateCount(h, count * num_ops),
+            qre.GateCount(cz, count * num_ops),
+            qre.GateCount(cnot, count * num_ops),
+        ]
+
+    @classmethod
+    def adjoint_resource_decomp(cls, target_resource_params: dict) -> list[GateCount]:
```

Comment:
Can you double check that the output makes sense in the edge case where select_swap_depth = 1 for both restored = True and restored = False. Can you add a test case for this.